framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/AbpDataAnnotationAutoLocalizationMetadataDetailsProvider.cs

```diff
@@ -54,7 +54,29 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
 
             displayMetadata.DisplayName = () =>
             {
-                var localizedString = localizer[PropertyLocalizationKeyPrefix + context.Key.Name];
+                /*
+                 * DisplayName:ClassName:PropertyName
+                 * DisplayName:PropertyName
+                 * ClassName:PropertyName
+                 * PropertyName
+                 */
+
+                LocalizedString localizedString = null;
+
+                if (context.Key.ContainerType != null)
+                {
+                    localizedString = localizer[PropertyLocalizationKeyPrefix + context.Key.ContainerType.Name + ":" + context.Key.Name];
```

Comment:
I am merging this assuming `context.Key` or `context.Key.ContainerType` can not be null, right?

Reply:
They are never `null` by default.